src/Events/Authorization/PublishScopeOrAccessTokenHandler.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Threading.Tasks;
+
+using Altinn.Platform.Events.Configuration;
+
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace Altinn.Platform.Events.Authorization
+{
+    /// <summary>
+    /// Authorization handler for the <see cref="PublishScopeOrAccessTokenRequirement"/> class.
+    /// </summary>
+    public class PublishScopeOrAccessTokenHandler : AuthorizationHandler<PublishScopeOrAccessTokenRequirement>
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PublishScopeOrAccessTokenHandler"/> class.
+        /// </summary>
+        public PublishScopeOrAccessTokenHandler(IHttpContextAccessor httpContextAccessor)
+        {
+            _httpContextAccessor = httpContextAccessor;
+        }
+
+        /// <summary>
+        /// Handles verification of requirements. Publish scope requirement or AccesToken requirement must be satisfied.
+        /// </summary>
+        protected override async Task<object> HandleRequirementAsync(AuthorizationHandlerContext context, PublishScopeOrAccessTokenRequirement requirement)
+        {
+            var authorizationService = _httpContextAccessor.HttpContext.RequestServices.GetService<IAuthorizationService>();
```

Comment:
Could we try to depend on IAuthorizationService directly instead of going through the HttpContextAccessor?

Reply:
Nope. it results in a circular dependency unfortunately..

Reply:
"A circular dependency was detected for the service of type 'Microsoft.AspNetCore.Authorization.IAuthorizationService'.
Microsoft.AspNetCore.Authorization.IAuthorizationService(Microsoft.AspNetCore.Authorization.DefaultAuthorizationService) -> Microsoft.AspNetCore.Authorization.IAuthorizationHandlerProvider(Microsoft.AspNetCore.Authorization.DefaultAuthorizationHandlerProvider) -> System.Collections.Generic.IEnumerable<Microsoft.AspNetCore.Authorization.IAuthorizationHandler> -> Microsoft.AspNetCore.Authorization.IAuthorizationHandler(Altinn.Platform.Events.Authorization.AccessTokenHandler) -> Microsoft.AspNetCore.Authorization.IAuthorizationService"

Reply:
hmm, okay. As we talked about. Fixing this probably requires changes in the AccessToken library. Maybe we should look into that?

Reply:
Actually, what if the new requirement inherit from AccessTokenRequirement. That way AccessTokenHandler would accept the new requirement as input. Then make a new Handler for the scope check that use the same requirement.

Could that work?

Reply:
I see one possible issue with my last idea. A requirement can't inherit from both ScopeAccessRequirement and AccessTokenRequirement. At least one of them would  need to be an interface.

Reply:
@SandGrainOne did you find a way to use multiple handlers, like we looked at on Wednesday?

Reply:
@b-brodie-digdir yes, we're currently testing changes in AccessTokenHandler and I've made a PR to update ScopeAccessHandler in the same way.